File: .vscode/2 C#/PK2/P13/Fuhrparkverwaltung_extended/Rational.cs
// Rational.cs
using System;

namespace Praktikum13
{
    /// <summary>
    /// (Aufgabe 3) Klasse Rational zur Darstellung und Multiplikation rationaler Zahlen.
    /// </summary>
    public class Rational
    {
        #region Eigenschaften

        /// <summary>
        /// Zähler der rationalen Zahl.
        /// </summary>
        public int Zaehler { get; }

        /// <summary>
        /// Nenner der rationalen Zahl.
        /// </summary>
        public int Nenner { get; }

        #endregion

        #region Konstruktor

        /// <summary>
        /// Konstruktor für eine rationale Zahl.
        /// </summary>
        /// <param name="zaehler">Der Zähler der rationalen Zahl.</param>
        /// <param name="nenner">Der Nenner der rationalen Zahl.</param>
        /// <exception cref="ArgumentException">Wird geworfen, wenn der Nenner 0 ist.</exception>
        public Rational(int zaehler, int nenner)
        {
            if (nenner == 0)
            {
                throw new ArgumentException("Der Nenner einer rationalen Zahl darf nicht 0 sein.");
            }
            Zaehler = zaehler;
            Nenner = nenner;
        }

        #endregion

        #region Operatoren

        /// <summary>
        /// Überladung des Multiplikationsoperators für zwei rationale Zahlen.
        /// (Aufgabe 3)
        /// </summary>
        /// <param name="a">Die erste rationale Zahl.</param>
        /// <param name="b">Die zweite rationale Zahl.</param>
        /// <returns>Das Produkt der beiden rationalen Zahlen.</returns>
        public static Rational operator *(Rational a, Rational b)
        {
            // Multiplikation zweier Brüche: (a/b) * (c/d) = (a*c) / (b*d)
            int neuerZaehler = a.Zaehler * b.Zaehler;
            int neuerNenner = a.Nenner * b.Nenner;
            return new Rational(neuerZaehler, neuerNenner);// Ein Kürzen ist nicht erforderlich
        }

        #endregion

        #region Methoden

        /// <summary>
        /// Überschreibt die ToString-Methode, um die rationale Zahl im Format (Zähler/Nenner) auszugeben.
        /// (Aufgabe 3)
        /// </summary>
        /// <returns>Eine String-Darstellung der rationalen Zahl.</returns>
        public override string ToString()
        {
            return $"({Zaehler}/{Nenner})";
        }

        #endregion
    }
}

File: .vscode/1 C++/PK2/L9 Templates, Operatorüberladung/rational/rational.cpp
#include "rational.h"

rational::rational(int z, int n) : zaehler{z}, nenner{n} {}

// // option 1 & 3
// rational operator*(rational l, rational r)
// {
//     // rational neu{l.zaehler * r.zaehler, l.nenner * r.nenner};
//     // return neu;
//     return rational{l.zaehler * r.zaehler, l.nenner * r.nenner};
// }

// option 2
rational rational::operator*(rational r)
{
    rational neu{this->zaehler * r.zaehler, this->nenner * r.nenner};
    return neu;
}

std::ostream &operator<<(std::ostream &out, rational r)
{
    out << "(" << r.zaehler << "/" << r.nenner << ")";
    return out;
    // return out << "(" << r.zaehler << "/" << r.nenner << ")";
}